core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -386,7 +386,12 @@ public void load(PluginWrapper wrapper) throws IOException {
             // initialize plugin
             try {
                 Plugin plugin = wrapper.getPlugin();
-                plugin.setServletContext(pluginManager.context);
+                if (plugin != null) {
+                    plugin.setServletContext(pluginManager.context);
+                }
+                else {
+                    throw new NullPointerException("Failed to initialize Plugin, because it is null");
```

Comment:
`because it is null` is redundant, you know it because we're throwing a `NullPointerException`. If you add the name of the plugin it may help administrators.

```suggestion
                    throw new NullPointerException("Failed to initialize the plugin " + wrapper.getDisplayName());
```